### PR TITLE
Mergify.yml merge to queue

### PR DESCRIPTION
**Pull request by nxei**
## What does this PR do?

It changes  merge action to queue in .mergify.yml

## Why is it important?

The configuration uses the deprecated strict mode of the merge action.
A brownout is planned for the whole December 6th, 2021 day.
This option will be removed on January 10th, 2022.
For more information: https://blog.mergify.io/strict-mode-deprecation/

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -1,3 +1,10 @@
+queue_rules:
```
Comment (v1v) [inline .mergify.yml:1]: Can you clarify if the queue_rules for open source projects support multiple entries?

Comment (nxei) [reply .mergify.yml:1]: Fixed.

**Comment by v1v** [comment]
Some of the git branches in this project are protected, and I've just seen an option called [`require_branch_protection`](https://docs.mergify.com/actions/queue/#id2), that option didn't exist in the `merge` action. Can you clarify if that's something that should be potentially enabled? And if so, what's the outcome?

File: .mergify.yml (hunk 1 above)
Comment (nxei) [reply .mergify.yml:1]: fixed

**Comment by nxei** [comment]
>Some of the git branches in this project are protected, and I've just seen an option called [`require_branch_protection`](https://docs.mergify.com/actions/queue/#id2), that option didn't exist in the `merge` action. Can you clarify if that's something that should be potentially enabled? And if so, what's the outcome?

<html><body>
<section id="id2">

require_branch_protection | bool | true | Whether branch protections are required for queueing pull requests. This option is ignored if the target queue has speculative_checks > 1.
-- | -- | -- | --

</section>
</body>
</html>

If I understand correctly it requires branch protections to be enabled in order pr to get in queue. By default it's enabled so for branches which have protections enabled configuration in .mergify.yml should be working.

**Review by v1v — APPROVED** [review]
Thanks for this and the clarifications around the `merge queue`!

File: .mergify.yml [hunk 2]
```diff
@@ -40,14 +44,14 @@ pull_request_rules:
         message: Automatically approving mergify
   - name: automatic squash and merge with success checks and the files matching the regex ^testing/environments/snapshot* are modified.
     conditions:
-      - check-success=beats-ci/pr-merge
       - label=automation
       - files~=^testing/environments/snapshot.*\.yml$
       - "#approved-reviews-by>=1"
+      - check-success=beats-ci/pr-merge
```
Comment (v1v) [inline .mergify.yml:50]: Can you add this check as used to be, at the beginning of the conditions?

**Review by v1v — CHANGES_REQUESTED** [review]
Can you revert the changes in the conditions to keep as used to be?

File: .mergify.yml (hunk 2 above)
Comment (nxei) [reply .mergify.yml]: moved - check-success=beats-ci/pr-merge to the begin of the list

File: .mergify.yml [hunk 3]
```diff
@@ -83,15 +87,15 @@ pull_request_rules:
         message: Automatically approving mergify
   - name: automatic squash and merge with success checks and the files matching the regex ^.mergify.yml is modified.
     conditions:
-      - check-success=beats-ci/pr-merge
       - label=automation
       - files~=^\.mergify\.yml$
       - head~=^add-backport-next.*
       - "#approved-reviews-by>=1"
+      - check-success=beats-ci/pr-merge
```
[the remark this answers is not in the document]
Comment (nxei) [reply .mergify.yml]: moved - check-success=beats-ci/pr-merge to the begin of the list

**Review by v1v — APPROVED** [review]
LGTM

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**